File: 1456_MaximumNumberOfVowelsInSubstringOfGivenLength/maximumNumberOfVowelsInSubstringOfGovenLength.py
def maxVowels(s: str, k: int) -> int:
    curV = 0
    maxV = 0
    leftPtr = 0

    for rightPtr in range(len(s)):
        if self.isVowel(s[rightPtr]):
            curV += 1
        k -= 1
        maxV = max(maxV, curV)

        if k <= 0:
            if self.isVowel(s[leftPtr]):
                curV -= 1
            leftPtr += 1
            k += 1
    return maxV

def isVowel(ch: str) -> bool:
    if ch == "a" or ch == "e" or ch == "i" or ch == "o" or ch == "u":
        return True
  
    return False

print(maxVowels("abciiidef", 3))
print(maxVowels("aeiou", 2))
print(maxVowels("leetcode", 3))

File: 1456_MaximumNumberOfVowelsInSubstringOfGivenLength/maximumNumberOfVowelsInSubstringOfGovenLength.js
var maxVowels = function(s, k) {
    let maxV = 0
    let curV = 0
    var leftPtr = 0

    for (let rightPtr = 0; rightPtr < s.length; rightPtr++) {
    	if (isVowel(s[rightPtr])) {
    		curV++;
    	}
    	k--;
    	maxV = Math.max(maxV, curV);

    	if (k <= 0) {
    		if (isVowel(s[leftPtr])) {
    			curV--;
    		}
    		leftPtr++;
    		k++;
    	}
    }
    return maxV
};


var isVowel = function(ch) { 
	if (ch == "a" || ch == "e" || ch == "i" || ch == "o" || ch == "u") {
		return true
  	}
  	return false
};

console.log(maxVowels("abciiidef", 3))
console.log(maxVowels("aeiou", 2))
console.log(maxVowels("leetcode", 3))